nntrainer/layers/layer_node.cpp

```diff
@@ -422,6 +422,10 @@ void LayerNode::read(std::ifstream &file) {
     /// @note shared weights are only be read at the first acecss
     if (run_context->isGradientLastAccess(i)) {
       run_context->getWeight(i).read(file);
+      // @note read optimizer variables
+      for (unsigned int j = 0; j < run_context->getNumWeightOptVar(i); ++j) {
```

Comment:
This PR will not provide backward compatibility and will break with existing saved model files. In my understanding, there are at least a couple of solutions:
1. loading the optimizer variables to be controlled by a property/config - which can be disabled by default providing backward compatibility to this PR
2. the optimizer variables are located at the end of the file, and must be loaded/saved after all the weights - this provides backward compatibility but requires lots of changes.

I would recommend to go with option 1.

Reply:
I think it is not the kind of feature to keep the backward compatibility. It's a bug. It might be better that we notice and encourage to change.

Reply:
This discussion is pretty much similar to the other comment, both realizing the need for a model file structure.
Also, I think with this change, we need to update our release version number as this is a breaking change.

Regarding model compatibility, an even easier option is to support a new enum BIN_V2 with this if needed.

Reply:
Right. I agree. But it is more than this pr is for.. and we need to deal with separate pr and issue. For temporal fixes, I will manage and update this PR with a small scope. Not a new release of NNTrainer or BIN_V2.